groups/validate.go

```diff
@@ -104,196 +134,232 @@ func TestMergedGroupsConfig(t *testing.T) {
 			if !ok || whoCanViewMembership != "ALL_MEMBERS_CAN_VIEW" {
-				t.Errorf("group '%s': must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW", g.Name)
+				log.Printf("group '%s': must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW", g.Name)
+				errs++
 			}
 		}
 	}
 	foundGKEGroup := false
-	for _, g := range cfg.Groups {
-		if g.EmailId == "gke-security-groups@kubernetes.io" {
+	for _, g := range groupsConfig.Groups {
+		if g.EmailId == "gke-security-groups@"+primaryDomain {
 			foundGKEGroup = true
 			// this is necessary for group-based rbac to work
 			whoCanViewMembership, ok := g.Settings["WhoCanViewMembership"]
 			if !ok || whoCanViewMembership != "ALL_MEMBERS_CAN_VIEW" {
-				t.Errorf("group '%s': must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW", g.Name)
+				log.Printf("group '%s': must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW", g.Name)
+				errs++
 			}
 			for _, email := range g.Members {
 				if _, ok := rbacEmails[email]; !ok {
-					t.Errorf("group '%s': invalid member '%s', must be a k8s-infra-rbac-*@kubernetes.io group", g.Name, email)
+					log.Printf("group '%s': invalid member '%s', must be a k8s-infra-rbac-*@"+primaryDomain+"group", g.Name, email)
+					errs++
 				} else {
 					rbacEmails[email] = true
 				}
 			}
 		}
 	}
 	if !foundGKEGroup {
-		t.Errorf("group '%s' is missing", "gke-security-groups@kubernetes.io")
+		log.Printf("group '%s' is missing", "gke-security-groups@"+primaryDomain)
+		errs++
 	}
 	for email, found := range rbacEmails {
 		if !found {
-			t.Errorf("group '%s': must be a member of gke-security-groups@kubernetes.io", email)
+			log.Printf("group '%s': must be a member of gke-security-groups@"+primaryDomain, email)
+			errs++
 		}
 	}
+
+	if errs > 0 {
+		os.Exit(1)
+	}
 }
 
 // Enforce conventions for SRC groups
 // - groups can't own other groups, so for groups that should be owned by
 //	 security@kubernetes.io should own, make sure the owners match
-func TestSecurityResponseCommitteeGroups(t *testing.T) {
+func VerifySecurityResponseCommitteeGroups(config Config) {
+	if config.SkipKubernetesIOTests {
+		return
+	}
+	errs := 0
 	pscGroups := []string{
 		"distributors-announce@kubernetes.io",
 		"security-discuss-private@kubernetes.io",
 	}
 	owners := []string{}
-	for _, g := range cfg.Groups {
+	for _, g := range groupsConfig.Groups {
 		if g.EmailId == "security@kubernetes.io" {
 			owners = g.Owners
 			break
 		}
 	}
 	for _, pscGroup := range pscGroups {
-		for _, g := range cfg.Groups {
+		for _, g := range groupsConfig.Groups {
 			if g.EmailId == pscGroup {
 				if !reflect.DeepEqual(owners, g.Owners) {
-					t.Errorf("group '%s': owners must match owners from security@kubernetes.io, expected: %v, actual: %v", pscGroup, owners, g.Owners)
+					log.Printf("group '%s': owners must match owners from security@kubernetes.io, expected: %v, actual: %v", pscGroup, owners, g.Owners)
+					errs++
 				}
 				break
 			}
 		}
 	}
```

Comment:
It may not matter, but it would be more efficient to put the `pscGroups` into a set or map and scan `groupsConfig.Groups` only once (or 1.5 times, since we scan for `security@kubernetes.io` earlier.

Separately, it feels like having `groupsConfig.Groups` be a `map[string]GroupConfig` would be a big win for _some_ of these loops, especially if you could combine the "verification function" for several different verifications using something like the Visitor pattern. This is definitely a TODO item, not a "do it in this PR" item.

Reply:
Added TODO note